File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/vgmrips/Pack.java
package app.zxtune.fs.vgmrips;

import androidx.annotation.NonNull;

// Backward links are not need
public class Pack {
  @NonNull
  public String id;
  @NonNull
  public String title;
  public int songs;
  public int score;
  public int ratings;

  Pack(String id, String title) {
    this.id = id;
    this.title = title;
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/vgmrips/Pack.kt
package app.zxtune.fs.vgmrips

// Backward links are not need
data class Pack(
    val id: Id,
    val title: String,
    // unescaped path after /files/
    val archive: FilePath,
    // unescaped path after /packs/images/large/
    val image: FilePath? = null,
    val songs: Int = 0,
    val size: String = "",
) {
    init {
        require(id.value.isNotEmpty())
        require(archive.value.isNotEmpty())
        image?.run {
            require(value.isNotEmpty())
        }
    }

    data class Id(val value: String)
}

data class FilePath(val value: String) {
    init {
        require(value.isNotEmpty() && '/' != value.first())
    }

    val displayName
        get() = value.substringAfterLast('/').substringBeforeLast('.')
}
